src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -482,6 +477,12 @@ public List<PackageManagerSearchElement> ListAll()
         {
             CachedPackageList = new List<PackageManagerSearchElement>();
 
+            // Attempt to load user votes prior to using it
+            if (AuthenticationManager.LoginState.Equals(LoginState.LoggedIn) && Uservotes == null)
+            {
+                Task.Run(() => this.Uservotes = this.Model.UserVotes());
+            }
+
             foreach (var header in Model.ListAll())
```

Comment:
Model.ListAll() will block the UI thread too.
Having only the `this.Uservotes = this.Model.UserVotes()` call on the threadPool will run the risk of concurrency issues (either with the `Uservotes` member or with RestSharp internal data)

I propose that you run the  `this.Uservotes = this.Model.UserVotes()` synchronously on the UI thread (no Task.Run) for now. It will mean slightly longer hang when showing the PackageManager window, but with no concurrency issues.  
@dnenov FYI

Reply:
So, this change will now mean that `uservotes()` runs in a blocking manner upon opening `PackageDetailsView` (in the side bar) instead of blocking the main `PackagemanagerClient` view, which it was doing earlier?  Is that right?

Reply:
Basically.
It was blocking dynamo startup (PackagemanagerClient view is initialized at dynamo startup)
Now the blocking code will run when showing the main pacakge manager window (but this is run in a separate thread asynchronously) 
Also will run in specific UI commands (like Upvoting, details etc) But there is a cache for this which will mitigate the cost of making more calls now (the code for it is upstream)